src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -393,9 +392,12 @@ public boolean authenticate(final SecurityRequestChannel request) {
             final User impersonatedUser = impersonate(request, authenticatedUser);
             final User effectiveUser = impersonatedUser == null ? authenticatedUser : impersonatedUser;
             threadPool.getThreadContext().putTransient(ConfigConstants.OPENDISTRO_SECURITY_USER, effectiveUser);
+            if (effectiveUser != authenticatedUser) {
+                threadPool.getThreadContext()
+                    .putTransient(ConfigConstants.OPENDISTRO_SECURITY_IMPERSONATION_INITIATING_USER, authenticatedUser.getName());
+            }
             UserSubject subject = new UserSubjectImpl(threadPool, effectiveUser);
             threadPool.getThreadContext().putPersistent(ConfigConstants.OPENDISTRO_SECURITY_AUTHENTICATED_USER, subject);
```

Comment:
Thank you @olkkoti . Apologies for not getting back sooner.

Curious, why not use the value from this persistent header?

In the case where impersonation is being used, looks like the current behavior is to audit log the authenticated user (not the user being impersonated - makes sense). In that case, maybe we can use the new header you are introducing in here and use it if the header is populated.

Reply:
Just triggered the PR checks to run. ~30 min for some of them to complete.

Reply:
Ok, another problem. To preserve the original functionality, we should do logSucceededLogin right after we have fetched security user, and only if it is set. Now at least the tests that I am able to run succesfully locally pass.